app/chat/page.tsx

```diff
@@ -1,19 +1,56 @@
 'use client'
-import { Suspense } from 'react'
+
+import { Suspense, useState } from 'react'
 import { Flex } from '@radix-ui/themes'
 import { Chat, ChatContext, ChatSideBar, useChatHook } from '@/components'
+import { WelcomeScreen } from '@/components/Welcome'
 import PersonaModal from './PersonaModal'
 import PersonaPanel from './PersonaPanel'
 
 const ChatProvider = () => {
   const provider = useChatHook()
+  const [showWelcome, setShowWelcome] = useState(true)
+
+  const handleSuggestionClick = (prompt: string) => {
+    // Create new chat if needed
+    if (!provider.currentChatRef?.current) {
```

Comment:
always create new chat from suggestion click. This avoid future problem with chat override